sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -71,4 +72,19 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     assert(!rawTable.properties.contains(HiveExternalCatalog.DATASOURCE_PROVIDER))
     assert(DDLUtils.isHiveTable(externalCatalog.getTable("db1", "hive_tbl")))
   }
+
+  test("alter table schema should save case-sensitive schema to properties") {
```

Comment:
Fair enough. My only concern here would be that ```alterTableSchema()``` is using a different code path to set the Hive table properties than ```createTable()``` does.